platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -91,7 +91,7 @@ public void putCandidateRoster(@NonNull final Roster candidateRoster) {
                 RosterUtils.hash(candidateRoster).getBytes();
 
         // update the roster state/map
-        final RosterState previousRosterState = rosterStateOrThrow();
+        final RosterState previousRosterState = rosterStateOrDefault();
```

Comment:
You're correct in any saved state the `RosterState` is non-null.

The behavior of returning `RosterState.DEFAULT` here will apply only at network genesis, and made it a bit easier to implement the `RosterService` schema genesis case.

Reply:
I see. Thanks for confirming that. As long as updates to the roster state are eventually persisted to the actual State after the genesis happens, I suppose this should be good. I'd still prefer to find a solution to avoid using the object in memory because if it gets used accidentally in a scenario where we don't expect it to be used, we'd rather get an exception than silently update a field in memory and potentially miss updating it on disk...